File: app/src/main/java/com/kieran/winnipegbusbackend/TripPlanner/classes/Segment.java
package com.kieran.winnipegbusbackend.TripPlanner.classes;

import com.kieran.winnipegbusbackend.TripPlanner.LocationFactory;

import org.json.JSONException;
import org.json.JSONObject;

public abstract class Segment {
    Times times;
    Location from;
    Location to;

    public Segment(JSONObject object, String fromTitle, String toTitle) {
        try {
            JSONObject times = object.getJSONObject("times");
            this.times = new Times(times);

            from = LocationFactory.createLocation(object.getJSONObject("from"), fromTitle);
            to = LocationFactory.createLocation(object.getJSONObject("to"), toTitle);

        }catch (JSONException ex) {

        }
    }

    @Override
    public String toString() {
        return from.getTitle() + " to " + to.getTitle();
    }

    public Times getTimes() {
        return times;
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/TripPlanner/classes/Segment.kt
package com.kieran.winnipegbusbackend.TripPlanner.classes

import com.kieran.winnipegbusbackend.TripPlanner.LocationFactory

import org.json.JSONException
import org.json.JSONObject

abstract class Segment(tripParameters: TripParameters, `object`: JSONObject) {
    lateinit var times: Times
        internal set
    internal var from: Location? = null
    internal var to: Location? = null

    init {
        try {
            val times = `object`.getJSONObject("times")
            this.times = Times(times)

            from = LocationFactory.createLocation(tripParameters, `object`.getJSONObject("from"))
            to = LocationFactory.createLocation(tripParameters, `object`.getJSONObject("to"))

        } catch (ex: JSONException) {

        }

    }

    override fun toString(): String {
        return from!!.title + " to " + to!!.title
    }
}
